_posts/2020-09-19-github-actions-with-faasd.md

```diff
@@ -0,0 +1,256 @@
+
+Create a DigitalOcean Droplet with a Ubuntu distribution, with any plan according to what you want to run, faasd being light weight I'm going to use the $5 basic plan.
+
+![Choose a distribution and a plan](/images/2020-09-19-github-actions-faasd/choose-distro.jpg)
+
+Under `Select additional options`, select `User data` and paste in the contents of the cloud-config file from [here](https://github.com/openfaas/faasd/blob/master/cloud-config.txt)
+
+![Select additional options](/images/2020-09-19-github-actions-faasd/additional-options-cloud-config.jpg)
+
+
+![faasd droplet](/images/2020-09-19-github-actions-faasd/faasd-droplet.jpg)
+
+Within as little as 30 seconds, your faasd instance will be up and running.
+
+## Get the OpenFaaS gateway password
+
+Get the gateway password by running the following commands
+
+```sh
+export IP="143.110.184.36"
+export PASSWORD=$(ssh root@$IP "sudo cat /var/lib/faasd/secrets/basic-auth-password")
+```
+
+Assuming that you have faas-cli already installed, login to faasd with faas-cli with the following commands
+
+```sh
+export OPENFAAS_URL=http://$IP:8080
+echo $PASSWORD | faas-cli login --password-stdin -u admin -g $OPENFAAS_URL
+```
+
+## Create an OpenFaaS function
+
+Now that we've logged into faasd with faas-cli, we're ready to run our functions.
+I'm going to deploy a golang function for this tutorial.
+
+```sh
+# create a new directory for our project
+mkdir faasd-hello
+# pull golang-http templates
+faas-cli template store pull golang-http
+# create a new function with faas-cli
+faas-cli new faasd-hello --lang golang-http --prefix utsavanand2
+```
+
+If you look into the root directory of the project, faas-cli has created two additional files:
+
+```
+./faasd-hello/handler.go
+./faasd-hello.yml
+```
+
+The content of our handler looks something like this:
+![./faasd-hello/handler.go](/images/2020-09-19-github-actions-faasd/handler-go.jpg)
+
+The YAML file is used to configure the deployment and the build and runtime environment of the function
+~[./faasd-hello.yml](/images/2020-09-19-github-actions-faasd/faasd-hello-yaml.jpg)
+
+## Setup a workflow file for Github Actions
+
+In the root directory of the project run:
+
+```sh
+mkdir -p .github/workflows
+nano ./github/workflows/main.yml
+```
+
+And paste in the following contents into main.yml file
+
+```yaml
+name: CI-CD
+
+on:
+  push:
+    branches: 
+      - master
+
+jobs:
+  func-build:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Pull template
+        run: |
+          docker run --rm -v `pwd`:`pwd` -w `pwd` \
+          openfaas/faas-cli:latest-root \
+          template store pull golang-http
+      - 
+        name: Running faas-cli skrinkwrap build
+        run: |
+          docker run --rm -v `pwd`:`pwd` -w `pwd` \
+          openfaas/faas-cli:latest-root \
+          build -f faasd-hello.yml --shrinkwrap
+      -
+        name: Login to DockerHub
```

Comment:
I would rather see you use the GitHub container registry for this ghcr.io because the Docker Hub is going over to a charging model and GitHub is free, and supports private containers.